silhouette/src/main/scala/silhouette/http/client/RequestBuilder.scala

```diff
@@ -0,0 +1,77 @@
+/**
+ * Licensed to the Minutemen Group under one or more contributor license
+ * agreements. See the COPYRIGHT file distributed with this work for
+ * additional information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package silhouette.http.client
+
+import silhouette.http.Cookie
+import scala.concurrent.Future
+
+/**
+ * Decorates a framework specific request builder implementation.
+ *
+ * @tparam R The type of request or request builder.
+ */
+protected[silhouette] trait RequestBuilder[R] {
+
+  /**
+   * Returns a copy of this instance with a new Uri.
+   * @param url The url to set.
+   * @return
+   */
+  def withUrl(url: String): RequestBuilder[R]
+
+  /**
+   * Returns a copy of this instance with a new method.
+   * @param method The method to set.
+   * @return
+   */
+  def withMethod(method: String): RequestBuilder[R]
+
+  /**
+   * Returns a copy of this instance with the list of headers set.
+   * @param params The headers to set.
+   * @return
+   */
+  def withHeader(params: Seq[(String, String)]): RequestBuilder[R]
+
+  /**
+   * Returns a copy of this instance with a new data.
+   * @param data The data to set.
+   * @return
+   */
+  def withData(data: String): RequestBuilder[R]
```

Comment:
Yes, we should implement a data extractor (or body extractor), but for now I've just keep it as string